src/main/java/io/phasetwo/service/auth/OrgAddUserAuthenticatorFactory.java

```diff
@@ -57,16 +61,34 @@ private void addUser(AuthenticationFlowContext context) {
       }
       if (!org.hasMembership(context.getUser())) {
         log.infof(
-            "granting membership to %s for user %s",
-            org.getName(), context.getUser().getUsername());
+                "granting membership to %s for user %s",
+                org.getName(), context.getUser().getUsername());
         org.grantMembership(context.getUser());
-        // TODO default roles from config??
+
+        orgs
+                .getUserInvitationsStream(context.getRealm(), context.getUser())
+                .filter(invitationModel -> invitationModel.getOrganization().getId().equals(org.getId()))
+                .findFirst()
+                .ifPresent(invitationModel -> addRolesFromInvitation(invitationModel, context.getUser()));
       }
     } else {
       log.infof("No organization owns IdP %s", brokerContext.getIdpConfig().getAlias());
     }
   }
 
+  void addRolesFromInvitation(InvitationModel invitation, UserModel user) {
```

Comment:
Is this the same as the method from the other class? Can we put this in a common location so both can use the same code?

Reply:
We cannot. The logic from the InvitationRequiredAction implies that assign user based on a invitation. 
In the case of this authenticator we always assign the user to a organization which has the Idp as a entrypoint, and then check if a invitation exists in order to assign the roles. 
If we would use the same method we would assign the organization based on the invitation